Source/WebCore/dom/EventLoop.cpp

```diff
@@ -406,6 +406,63 @@ Markable<MonotonicTime> EventLoop::nextTimerFireTime() const
     return m_nextTimerFireTimeCache;
 }
 
+class JSMicrotaskDispatcher final : public JSC::MicrotaskDispatcher {
+    WTF_MAKE_TZONE_ALLOCATED(JSMicrotaskDispatcher);
+public:
+
+    JSMicrotaskDispatcher(EventLoopTaskGroup& group)
+        : m_group(group)
+    {
+    }
+
+    ~JSMicrotaskDispatcher() final = default;
+
+    JSC::QueuedTask::Status run(JSC::QueuedTask& task) final
+    {
+        auto group = m_group.get();
```

Comment:
Hm... this code is repeated like 3 different times. Can we share the code somehow? Maybe we can have something like `WebCoreMicrotaskDispatcher` which knows what to return as a result?